### PR TITLE
Refactor kprobe tracing functionality out of system/socket

### PR DESCRIPTION
**Pull request by adriansr**
## What does this PR do?

Refactors the high-level kprobe-tracing functionality out of Auditbeat's _system/socket_ into `x-pack/auditbeat/tracing/kprobes`.

## Why is it important?

This makes it easier to build new functionality based on kprobe tracing.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Example

An example of reusing this code: 
https://github.com/adriansr/beats/tree/poc_auditd_syscall_overhead/x-pack/auditbeat/module/auditd/monitoring/

### CODE REVIEW

File: x-pack/auditbeat/module/system/socket/kprobes.go [hunk 1]
```diff
@@ -8,134 +8,19 @@ package socket
...
-					filter, probe.Probe.Filter)
-			}
-		}
-		return probe
-	}
-}
-
 // KProbes shared with IPv4 and IPv6.
-var sharedKProbes = []helper.ProbeDef{
+var sharedKProbes = []kprobes.ProbeDef{
```
Comment (andrewstucki) [inline x-pack/auditbeat/module/system/socket/kprobes.go:23]: Though not necessary in this commit, it'd probably be worth splitting these into their own files similar to the other `guess`-based files, just for easier management.

File: x-pack/auditbeat/module/system/socket/guess/deref.go [hunk 2]
```diff
@@ -79,22 +79,22 @@ func (g *guessDeref) Requires() []string {
 
 // Probes returns a kprobe on uname() that dumps the first bytes
 // pointed to by its first parameter.
-func (g *guessDeref) Probes() ([]helper.ProbeDef, error) {
-	return []helper.ProbeDef{
+func (g *guessDeref) Probes() ([]kprobes.ProbeDef, error) {
+	return []kprobes.ProbeDef{
 		{
 			Probe: tracing.Probe{
```
Comment (andrewstucki) [inline x-pack/auditbeat/module/system/socket/guess/deref.go:85]: wondering why keep the package different for `kprobes` v. `tracing`? Are we planning on doing some other types of tracing (i.e. tracepoint, uprobe, etc?)

Comment (adriansr) [reply x-pack/auditbeat/module/system/socket/guess/deref.go:84]: Not really. Makes sense to merge the packages.

**Comment by adriansr** [comment]
Turning it into a draft for now as it needs more changes:
- Rebase on top of current socket dataset
- Better guess instantiation to avoid globals

**Pull request closed without merging**